ios/RNAudioRecorderPlayer.swift

```diff
@@ -566,4 +570,34 @@ class RNAudioRecorderPlayer: RCTEventEmitter, AVAudioRecorderDelegate {
             return "audio"
         }
     }
+
+    /**********    Recorder Helpers (tracking recording duration)    **********/
+
+    private func recordingDidStart() {
+        self.accumulatedRecordingDuration = 0
+        self.lastResumeTime = CACurrentMediaTime()
+    }
+
+    private func recordingDidPause() {
+        guard let lastResumeTime else { return }
+
+        self.accumulatedRecordingDuration += CACurrentMediaTime() - lastResumeTime
+        self.lastResumeTime = nil
+    }
+
+    private func recordingDidResume() {
+        self.lastResumeTime = CACurrentMediaTime()
+    }
+
+    private func recordingDidFinish() {
+        self.accumulatedRecordingDuration = 0
+        self.lastResumeTime = nil
```

Comment:
Feels like we should do

```swift
self.accumulatedRecordingDuration += CACurrentMediaTime() - lastResumeTime
self.lastResumeTime = nil
```

This way, we can finish the recording and still access the duration